src/auth.test.ts

```diff
@@ -0,0 +1,219 @@
+  const AuthData = bb.generateAuthData();
+
+  const expectedUrl: string =
+    `${BASE_AUTH_URL}?client_id=${CLIENT_ID}` +
+    `&redirect_uri=${CALLBACK_URL}&state=${AuthData.state}` +
+    `&response_type=code&code_challenge_method=S256&code_challenge=${AuthData.codeChallenge}`;
+
+  const url = bb.generateAuthorizeUrl(AuthData);
+
+  expect(url).toBe(expectedUrl);
+});
+
+test("expect auth method generateTokenPostData() function", () => {
+  const AuthData = bb.generateAuthData();
+
+  const expectedPostData = {
+    client_id: "foo",
+    client_secret: "bar",
+    code: "test-code",
+    code_challenge: AuthData.codeChallenge,
+    code_verifier: AuthData.verifier,
+    grant_type: "authorization_code",
+    redirect_uri: "http://localhost/callback/",
+  };
+
+  const postData = generateTokenPostData(bb, AuthData, "test-code");
+
+  expect(postData).toStrictEqual(expectedPostData);
+});
+
+describe("auth method getAuthorizationToken", () => {
+  const authData = bb.generateAuthData();
+
+  const BB2_ACCESS_TOKEN_URL = bb.baseUrl + "/" + bb.version + "/o/token/";
+
+  const mockResponse = {
+    status: 200,
+    data: {
+      access_token: "66ClP4JCjpdxmkC6bEKYQFLOWXnraJ",
+      expires_in: 36000,
+      token_type: "Bearer",
+      scope: [
+        "introspection",
+        "patient/Coverage.read",
+        "patient/ExplanationOfBenefit.read",
+        "patient/Patient.read",
+        "profile",
+      ],
+      refresh_token: "jV3knA4xmZ1enK8Rg1Lub5hmIsc9Ad",
+      patient: "-20140000000051",
+      expires_at: 1646195004.3654683,
+    },
+  };
+
+  const mockResponseMissingExpiresAt = {
+    status: 200,
+    data: {
+      access_token: "66ClP4JCjpdxmkC6bEKYQFLOWXnraJ",
+      expires_in: 36000,
+      token_type: "Bearer",
+      scope: [
+        "introspection",
+        "patient/Coverage.read",
+        "patient/ExplanationOfBenefit.read",
+        "patient/Patient.read",
+        "profile",
+      ],
+      refresh_token: "jV3knA4xmZ1enK8Rg1Lub5hmIsc9Ad",
+      patient: "-20140000000051",
+    },
+  };
+
+  it("expect successful response", async () => {
+    mockedAxios.post.mockResolvedValueOnce(mockResponse);
+
+    await expect(async () => {
+      const expectedAuthToken = new AuthorizationToken(mockResponse.data);
+
+      const ret = await bb.getAuthorizationToken(
+        authData,
+        "test-code",
+        authData.state,
+        "test-error"
+      );
+
+      expect(ret).toEqual(expectedAuthToken);
+    }).not.toThrow(Error);
+
+    expect(axios.post).toHaveBeenCalledWith(
+      BB2_ACCESS_TOKEN_URL,
+      expect.anything()
+    );
+  });
+
+  it("expect successful response with missing expires_at", async () => {
+    mockedAxios.post.mockResolvedValueOnce(mockResponseMissingExpiresAt);
+
+    await expect(async () => {
+      const expectedAuthToken = new AuthorizationToken(
+        mockResponseMissingExpiresAt.data
+      );
+
+      const ret = await bb.getAuthorizationToken(
```

Comment:
I think since `getAuthorizationToken` has a return value, we'll make sure the return value is valid. Checking that axios got the right input is a good test as well but then what the code does with the results of that we'll also need to validate via the return object.

Reply:
@nbragdon Added some checks for the return on `getAuthorizationToken()`.